### PR TITLE
ROX-12480: run terraform script from a github action

### PR DESCRIPTION
**Pull request by porridge**
## Checklist (Definition of Done)

- [x] ~Unit and integration tests added~
- [x] Added test description under `Test manual`
- [x] ~Evaluated and added CHANGELOG.md entry if required~
- [x] ~Documentation added if necessary (i.e. changes to dev setup, test execution, ...)~ :point_right: planning to do docs overhaul once this is well tested and applies to production too
- [ ] CI and all relevant tests are passing
- [x] Add the ticket number to the PR title if available, i.e. `ROX-12345: ...`
- [x] ~Discussed security and business related topics privately. Will move any security and business related topics that arise to private communication channel.~

## Test manual

The script is already [tested in a previous PR](https://github.com/stackrox/acs-fleet-manager/pull/505), but no good way to test the integration before the actual push.

### CODE REVIEW

File: .github/workflows/deploy-stage.yaml [hunk 1]
```diff
@@ -29,10 +29,10 @@ jobs:
...
           role-to-assume: arn:aws:iam::${{ secrets.AWS_ACCOUNT_ID }}:role/github
-      - name: List Parameter Store services
+      - name: Run terraforming THE on stage cluster
         env:
           USE_AWS_VAULT: false
         run: |
-          source ./scripts/lib/external_config.sh
-          init_chamber
-          run_chamber list-services
+          set -euo pipefail
```
Comment (kovayur) [inline .github/workflows/deploy-stage.yaml:36]: nit: for the below commands this should be enough.
```suggestion
          set -e
```

File: .github/workflows/deploy-stage.yaml [hunk 2]
```diff
@@ -29,10 +29,10 @@ jobs:
         with:
           aws-region: ${{ secrets.AWS_REGION }}
           role-to-assume: arn:aws:iam::${{ secrets.AWS_ACCOUNT_ID }}:role/github
-      - name: List Parameter Store services
+      - name: Run terraforming THE on stage cluster
```
Comment (kovayur) [inline .github/workflows/deploy-stage.yaml:32]: "THE" is in the right place? I'm not sure I understand the meaning. 😅

**Review by kovayur — APPROVED** [review]
All good. Just minor remarks, they can be ignored

**Comment by kovayur** [comment]
Thinking about the change another minute, I see that there were no changes introduced for `FLEETSHARD_SYNC_IMAGE` variable. It means that we'll redeploy the same image on each merge. Do you plan to change it in a future?

**Comment by porridge** [comment]
FTR, this is waiting for:
- https://github.com/stackrox/acs-fleet-manager/pull/550
- https://github.com/stackrox/acs-fleet-manager/pull/548 (not strictly required for staging, but better safe than sorry)

**Comment by porridge** [comment]
> Thinking about the change another minute, I see that there were no changes introduced for the `FLEETSHARD_SYNC_IMAGE` variable. It means that we'll redeploy the same image on each merge. Do you plan to change it in a future?

Yes, I'm taking baby steps for now.

File: .github/workflows/deploy-stage.yaml (hunk 2 above)
Comment (porridge) [reply .github/workflows/deploy-stage.yaml]: Fixed. The capitalization was a poor joke, hinting at the fact that in the future there might be more clusters, especially in production, and we'll need to figure out how to process them all.

File: .github/workflows/deploy-stage.yaml (hunk 1 above)
Comment (porridge) [reply .github/workflows/deploy-stage.yaml:36]: Yeah, I'm being "future-proof" :sweat_smile:

Comment (porridge) [reply .github/workflows/deploy-stage.yaml:36]: I mean, scripts tend to grow organically and people (myself included) often forget to extend the range of useful checks.

**Review by kovayur — APPROVED** [review]

**Comment by porridge** [comment]
Force-merging since the e2e test is irrelevant here, and the cause was an infrastructure problem anyway.

**Pull request merged**